docs/source/array-best-practices.rst

```diff
@@ -0,0 +1,91 @@
+Best Practices
+==============
+
+It is easy to get started with Dask arrays, but using them *well* does require
+some experience.  This page contains suggestions for best practices, and
+includes solutions to common problems.
+
+Use NumPy
+---------
+
+If your data fits comfortably in RAM and you are not performance bound, then
+using NumPy might be the right choice.  Dask adds another layer of complexity
+which may get in the way.
+
+
+Select a good chunk size
+------------------------
+
+A common performance problem among Dask Array users is that they have chosen a
+chunk size that is either too small (leading to lots of overhead) or poorly
+aligned with their data (leading to inefficient reading).
+
+While optimal sizes and shapes are highly problem specific, it is rare to see
+chunk sizes below 100 MB in size.  If you are dealing with float64 data then
+this is around ``(4000, 4000)`` in size.
+
```

Comment:
I think it would also be useful to add a section on the number of chunks. For very large arrays, the chunk size will need to be larger to keep the number of tasks <~100k. @mrocklin, is there a best practices section on # of tasks/chunks elsewhere we could link to?

Reply:
Joe see https://github.com/mrocklin/dask/pull/9

Reply:
I'd like to handle this in a general purpose Best Practices document in a follow-up PR

Reply:
How are you planning to handle it? Generally talking about how many tasks can be reasonably included in the task graph? Or something else?

Reply:
See #4745 

I would talk about general thoughts on overhead, and encourage people to use larger chunks.

Reply:
SGTM. Thanks for clarifying.